### PR TITLE
Allow changing default UI service + change String default UI service allowing log level control

### LINKED ISSUE #1741: Liquibase pollutes System.out in a Spring Boot application
Liquibase writes directly to System.out instead of writing through a configured logger in a Spring Boot Application. This is unexpected to the consumer of System.out, which expects, for example, JSON in System.out.

Liquibase version: 4.3.1
Spring boot version: 2.4.3
Message: Skipping auto-registration

Stack trace:

```
	at liquibase.ui.ConsoleUIService.sendMessage(ConsoleUIService.java:45)
	at liquibase.hub.HubUpdater.register(HubUpdater.java:372)
	at liquibase.Liquibase.lambda$update$1(Liquibase.java:242)
	at liquibase.Scope.lambda$child$0(Scope.java:160)
	at liquibase.Scope.child(Scope.java:169)
	at liquibase.Scope.child(Scope.java:159)
	at liquibase.Scope.child(Scope.java:138)
	at liquibase.Liquibase.runInScope(Liquibase.java:2322)
	at liquibase.Liquibase.update(Liquibase.java:216)
	at liquibase.Liquibase.update(Liquibase.java:202)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:322)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:275)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1153)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:907)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:582)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:144)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:767)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:426)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:326)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1311)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1300)
```

image

The only property related to liquibase in Spring Boot configuration:

```yaml
spring:
  liquibase:
    change-log: classpath:db-changelog.xml
```

### LINKED ISSUE #2396: Disable console logs for every changeset
## Environment

**Liquibase Version**: 4.7.1

**Liquibase Integration & Version**: <Pick one: CLI, maven, gradle, spring boot, servlet, etc.>

**Liquibase Extension(s) & Version**: 

**Database Vendor & Version**:

**Operating System Type & Version**:

## Description

Since 4.7.1 every changeset is logged to SOUT, and it seems like there is no way to turn it off. For us that means that tests logs are complete unusable during to hight amount of tests & changesets. 

## Steps To Reproduce

Just run the migration.

## Actual Behavior

For every change set a message in system out is appeared: 

```
Running Changeset: db/changelog/db.changelog-file.xml::version::author
```

## Expected/Desired Behavior

No message in SOUT. A way to completely replace `ConsoleUIService` with `LoggerUIService` so every message could be controlled by logger implementation

### PR DESCRIPTION
**Pull request by filipelautert**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description
This PR allows people to easily switch from ConsoleUiService to LoggerUIService by:
* Adding a new command line parameter called `uiService` that defaults to `CONSOLE` (current `ConsoleUiService`) but also accepts `LOGGER `(`LoggerUIService`) as a value
* Creates a new Spring property called `uiService` that accepts the same 2 values, but
* Switches to `LoggerUIService `as the default `UIService `for `LiquibaseSpring`

## Things to be aware of
* Uses `LoggerUIService `as the default `UIService `for `LiquibaseSpring`

## Things to worry about
* uiService property needs to be added to Spring project (an upstream PR needs to follow) - besides I doubt that Spring users will want to switch back to Console output. 

## Additional Context
Fixes #1741, Fixes #2396 

Summary is being handled on other 2 PR's: https://github.com/liquibase/liquibase/issues/4573 and https://github.com/liquibase/liquibase/pull/4574

### CODE REVIEW

File: liquibase-standard/src/main/java/liquibase/GlobalConfiguration.java [hunk 1]
```diff
@@ -221,6 +224,11 @@ public class GlobalConfiguration implements AutoloadedConfigurations {
                 .setDescription("Will perform xsd validation of XML changelog files. When many XML changelog files are included this validation may impact Liquibase performance. Defaults to true.")
                 .setDefaultValue(true)
                 .build();
+
+        UI_SERVICE = builder.define("uiService", UIServiceEnum.class)
+                .setDescription("Changes the default UI Service Logger used by Liquibase. Option are CONSOLE or LOGGER - defaults to CONSOLE.")
```
Comment (StevenMassaro) [inline liquibase-standard/src/main/java/liquibase/GlobalConfiguration.java:229]:
```suggestion
                .setDescription("Changes the default UI Service Logger used by Liquibase. Options are CONSOLE or LOGGER.")
```

Comment (StevenMassaro) [reply liquibase-standard/src/main/java/liquibase/GlobalConfiguration.java:229]: The default value automatically shows up in the console output, no need to repeat it in the description.

File: liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package liquibase.ui;
+
+public enum UIServiceEnum {
```
Comment (StevenMassaro) [inline liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java:3]: Please format this file, it's indented wrong.

Also, please add some javadoc.

**Review by StevenMassaro — CHANGES_REQUESTED** [review]

File: liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java (hunk 2 above)
Comment (filipelautert) [reply liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java:3]: + lombok

File: liquibase-standard/src/main/java/liquibase/GlobalConfiguration.java [hunk 3]
```diff
@@ -221,6 +224,11 @@ public class GlobalConfiguration implements AutoloadedConfigurations {
                 .setDescription("Will perform xsd validation of XML changelog files. When many XML changelog files are included this validation may impact Liquibase performance. Defaults to true.")
                 .setDefaultValue(true)
                 .build();
+
+        UI_SERVICE = builder.define("uiService", UIServiceEnum.class)
+                .setDescription("Changes the default UI Service Logger used by Liquibase. Options are CONSOLE or LOGGER.")
+                .setDefaultValue(UIServiceEnum.CONSOLE)
+                .build();
```
Comment (StevenMassaro) [inline liquibase-standard/src/main/java/liquibase/GlobalConfiguration.java:231]: One more thing here, I recommend you use the value handler util.

For an example, search for `ValueHandlerUtil.getEnum`.

Long term the command framework should automatically do this when using an enum.

File: liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package liquibase.ui;
+
+import lombok.Getter;
+
+/**
+ * Enum used to define the available UIServices provided by Liquibase that can be set using global parameters
+ */
+@Getter
```
Comment (StevenMassaro) [inline liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java:8]: If you really want to lombokify you can could also do `@AllArgsConstructor` or maybe just `@Constructor`

**Review by StevenMassaro — APPROVED** [review]

File: liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java (hunk 4 above)
Comment (filipelautert) [reply liquibase-standard/src/main/java/liquibase/ui/UIServiceEnum.java:8]: Nice, this one wasn't suggested by Sonar

File: liquibase-standard/src/main/java/liquibase/GlobalConfiguration.java (hunk 3 above)
Comment (filipelautert) [reply liquibase-standard/src/main/java/liquibase/GlobalConfiguration.java]: that's interesting - I searched for it and found out that it only exists in tests (?) - seems it was used at UpdateCommandStep but was removed. Anyway here is 
 it:  https://github.com/liquibase/liquibase/pull/4584/commits/718d57d3e6ebd22cacf63022bb275c66bdae9592 .

**Comment by filipelautert** [comment]
@tati-qalified do you think you can help me to test this PR? I can walkthrough the changes and expected results with you.

**Review by tati-qalified — APPROVED** [review]

**Comment by rberezen** [comment]
@filipelautert could you please take a look why tests fail? TIA

**Comment by filipelautert** [comment]
> @filipelautert could you please take a look why tests fail? TIA

@rberezen fixed 👍

**Review by suryaaki2 — APPROVED** [review]

**Comment by rberezen** [comment]
Investigating Functional Test Failure, please don't merge yet

**Comment by filipelautert** [comment]
@rberezen all green!

**Review by rberezen — APPROVED** [review]

**Comment by rberezen** [comment]
> @rberezen all green!

thanks! approved

**Comment by adrian-velonis1** [comment]
Documented: https://datical.atlassian.net/browse/PD-3925

When the docs are live, the URL will be: https://docs.liquibase.com/parameters/ui-service.html

**Pull request merged**